src/adapter/src/coord/sequencer.rs

```diff
@@ -1994,7 +1991,7 @@ impl Coordinator {
             .vars()
             .iter()
             .chain(catalog.system_config().iter())
-            .filter(|v| !v.experimental() && v.visible(session.user()))
+            .filter(|v| !v.experimental() && v.visible(session.user() == &*SYSTEM_USER))
```

Comment:
This is subjective, so feel free to ignore, it may be cleaner to add a method like the following to `User`:
```Rust
fn is_system_user(&self) -> bool {
  self == &*SYSTEM_USER
}
```

and use that here and below.